src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
```

Comment:
Using the reduce() method: Using the reduce() method allows you to conveniently calculate the sum immediately without having to manually create the sum variable and use the forEach() loop. The reduce() method passes the initial value (in this case 0) and accumulates the sum by going through each element of the men array. This is a more declarative approach that makes the code easier to understand.

<pre>
const sum = men.reduce((total, m) => (
  total + (m.died - m.born)
), 0);
</pre>